edb/lang/common/devmode.py

```diff
@@ -0,0 +1,94 @@
+#
+# This source file is part of the EdgeDB open source project.
+#
+# Copyright 2011-present MagicStack Inc. and the EdgeDB authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+import hashlib
+import os
+import pathlib
+import pickle
+import tempfile
+import typing
+
+
+def enable_dev_mode():
+    os.environ['__EDGEDB_DEVMODE'] = '1'
+
+
+def is_in_dev_mode() -> bool:
+    devmode = os.environ.get('__EDGEDB_DEVMODE', '0')
+    return devmode.lower() not in ('0', '', 'false')
+
+
+def get_dev_mode_cache_dir() -> os.PathLike:
+    if is_in_dev_mode():
+        root = pathlib.Path(__file__).parent.parent.parent.parent
+        cache_dir = (root / 'build' / 'cache')
```

Comment:
Let's ensure that `root / 'build'` exists (it must if you're in the dev mode).  This would help us when we move `edb.lang.common` to `edb.common`.